marlowe-dashboard-client/src/Welcome/State.purs

```diff
@@ -226,11 +195,41 @@ toWalletNicknameInput ::
   Functor m =>
   HalogenM (InputField.State WalletNicknameError) (InputField.Action WalletNicknameError) slots msg m Unit ->
   HalogenM State Action slots msg m Unit
-toWalletNicknameInput = mapSubmodule _walletNicknameInput WalletNicknameInputAction
+toWalletNicknameInput =
+  mapMaybeSubmodule
+    (_modal <<< _Just <<< _1 <<< _UseNewWallet <<< _1)
+    WalletNicknameInputAction
+    InputField.dummyState
+  where
+  _UseNewWallet =
+    prism' (uncurry UseNewWallet) case _ of
+      UseNewWallet id nickname -> Just $ Tuple id nickname
+      _ -> Nothing
```

Comment:
nice :smile:

Reply:
Lol, I'd rather move this into its own component than do this traversal gymnastics :P